Feat: 팀원 모두 참여한 초기 Swagger

zzoe2346 opened the pull request:
## #️⃣ 연관된 이슈
#2 

## 📝 작업 내용
- controller, dto 코드 구현
- Swagger setting
- dto의 getter는 스웨거를 위한 것

### 스크린샷 (선택)

## 💬 리뷰 요구사항(선택)

## ⏰ 현재 버그

## ✏ Git Close 
close #2

zzoe2346 committed:
Feat: 팀원 모두 참여한 초기 Swagger

- controller, dto 코드 구현
- Swagger setting
- dto의 getter는 스웨거를 위한 것

2iedo commented on src/main/java/com/example/sinitto/config/SwaggerConfig.java:
이건 무슨 코드인가요?? 스웨거 초기 화면인가요?

GitJIHO replied:
스웨거 작성을 위해 필요한 config로 알고 있습니다

zzoe2346 replied:
넵 초기화면 관련입니다. 제가 극 초기에 만들어놓고 그대로 둔거같아요.

2iedo commented on src/main/java/com/example/sinitto/controller/GuardGuidelineController.java:
카테고리 데이터베이스를 title, content로 하자고 했으니 카테고리에 해당하는 모든 가이드라인 조회는 필요없을 것 같아요

GitJIHO replied:
살짝 헷갈리는데 카테고리별로 여러개의 가이드라인 content 객체가 존재하는게 아니라 그냥 여러개의 정보를 하나의 content안에 모두 넣는걸로 하는걸까요?

2iedo replied:
네네 그냥 카페 게시물처럼 content 안에 모든 내용이 들어가는 걸로 이해하고 있습니다.

2iedo commented on src/main/java/com/example/sinitto/controller/HelloCallController.java:
서비스 수락하기의 request인 HelloCallAcceptRequest 클래스에 Long 하나만 있어서 굳이 클래스로 만들 필요 없을 것 같습니다.

2iedo replied:
HelloCallCompleteRequest도 위와 동일하여 클래스로 안만들어도 될 것 같습니다.

2iedo replied:
HelloCallCancelRequest도 위와 동일하여 클래스로 안만들어도 될 것 같습니다.

2iedo replied:
이것도 body에 담기 위해 클래스를 만든 것 같은데, 아래 의견과 동일합니다.

GitJIHO replied:
이게 swagger에서 request에 대한 정보를 표시하기 위해서는 dto상에 직접 어노테이션을 통해 정보를 주입해야 하는걸로 알고 있어서 삭제시 request에 대한 정보를 swagger에 제공할 다른 방법이 있을까요??

GitJIHO replied:
<img width="700" alt="image" src="https://github.com/user-attachments/assets/962856bc-9ecf-4a35-aba7-1dcbb1ce2239">

요런식으로 필드값이 하나라도 swagger때문에 step2에서 강제로 넣긴 했었습니다

2iedo commented on src/main/java/com/example/sinitto/controller/CallbackController.java:
CallbackCompleteRequest, CallbackAcceptRequest, CallbackCancelRequest 클래스도 CallbackId 하나만 있어 통일성을 위해 굳이 클래스로 만들 필요 없을 것 같습니다.

2iedo replied:
이게 body에 넣기 위해서 클래스로 만들었던 것으로 기억하는데, 방법의 차이라 저는 개인적으로 상세보기와 형식을 맞추는 것도 괜찮다고 생각합니다. body에 담기 위해 클래스 수가 너무 많아지는 느낌이네요

zzoe2346 replied:
아! 클래스 이렇게 생성한 이유가

그때 도훈님이랑 같이 만들때 그냥 `@RequestParam Long CallbackId` 이걸로 하였고 나머지도 다 이런식으로 바꿀수 있는건 수정하는데, **이렇게하면 당장 Swagger에서 이걸 인식을 못하더라구요**. 그래서 어쩔수 없이 클래스를 각각 만들었었습니다. 

미리 설명드렸어야되는데 죄송해요!
일단 Swagger에 표시가 되는걸 우선했습니다.

GitJIHO replied:
해당 의견은 동의합니다. 차라리 path variable로 모두 넣는다면 클래스가 줄 것 같기는 하네요. 논의가 필요해보입니다

2iedo reviewed (changes requested):
#3 코드 리뷰입니다.

zzoe2346 committed:
Refactor: HelloCallController '요청' 관련 리팩토링(RequestBody->PathVariable)

- RequestBody로 helloCallId 받던걸 PathVariable로 받도록 변경
- 쓸모 없어진 클래스 제거

zzoe2346 committed:
Refactor: CallbackController '요청' 관련 리팩토링(RequestBody->PathVariable)

- RequestBody로 callbackId 받던걸 PathVariable로 받도록 변경
- 쓸모 없어진 클래스 제거

zzoe2346 committed:
Refactor: PointRequest 로 통일

- PointWithdrawRequest -> PointRequest
- 둘 다 `int price;`만 가지고 있음
- 쓸모 없어진 클래스 제거
